fortitude/src/options.rs

```diff
@@ -0,0 +1,287 @@
+    /// Options are "off" (default), "ascii", and "fancy"
+    #[option(
+        default = "off",
+        value_type = "str",
+        scope = "progress-bar",
+        example = r#"
+          # Enable unicode progress bar
+          progress-bar = "fancy"
+       "#
+    )]
+    pub progress_bar: Option<ProgressBar>,
+
+    // Rule selection
+    /// A list of rule codes or prefixes to ignore. Prefixes can specify exact
+    /// rules (like `T003` or `superfluous-implicit-none`), entire categories
+    /// (like `T` or `typing`), or anything in between.
+    ///
+    /// When breaking ties between enabled and disabled rules (via `select` and
+    /// `ignore`, respectively), more specific prefixes override less
+    /// specific prefixes.
+    #[option(
+        default = "[]",
+        value_type = "list[RuleSelector]",
+        example = r#"ignore = ["superfluous-implicit-none"]"#
+    )]
+    pub ignore: Option<Vec<RuleSelector>>,
+
+    // TODO: fix default and example when default rules decided!
+    /// A list of rule codes or prefixes to enable. Prefixes can specify exact
+    /// rules (like `T003` or `superfluous-implicit-none`), entire categories
+    /// (like `T` or `typing`), or anything in between.
+    ///
+    /// When breaking ties between enabled and disabled rules (via `select` and
+    /// `ignore`, respectively), more specific prefixes override less
+    /// specific prefixes.
+    #[option(
+        default = r#"["E4", "E7", "E9", "F"]"#,
+        value_type = "list[RuleSelector]",
+        example = r#"
+            # On top of the defaults (`E4`, E7`, `E9`, and `F`), enable flake8-bugbear (`B`) and flake8-quotes (`Q`).
+            select = ["E4", "E7", "E9", "F", "B", "Q"]
+        "#
+    )]
+    pub select: Option<Vec<RuleSelector>>,
+
+    // TODO: fix default and example when default rules decided!
+    /// A list of rule codes or prefixes to enable, in addition to those
+    /// specified by [`select`](#check_select).
+    #[option(
+        default = "[]",
+        value_type = "list[RuleSelector]",
+        example = r#"
+            # On top of the default `select` (`E4`, E7`, `E9`, and `F`), enable flake8-bugbear (`B`) and flake8-quotes (`Q`).
+            extend-select = ["B", "Q"]
+        "#
+    )]
+    pub extend_select: Option<Vec<RuleSelector>>,
+
+    // File resolver options
+    /// A list of file extensions to check
+    #[option(
+        default = r#"["f90", "F90", "f95", "F95", "f03", "F03", "f08", "F08", "f18", "F18", "f23", "F23"]"#,
+        value_type = "list[str]",
+        example = r#"["f90", "fpp"]"#
+    )]
+    pub file_extensions: Option<Vec<String>>,
+
+    /// A list of file patterns to exclude from formatting and linting.
+    ///
+    /// Exclusions are based on globs, and can be either:
+    ///
+    /// - Single-path patterns, like `.mypy_cache` (to exclude any directory
+    ///   named `.mypy_cache` in the tree), `foo.f90` (to exclude any file named
```

Comment:
Should we use something other than `.mypy_cache` here? Maybe something like `generated`, which might contain Fortran code generated at compile time?